script/DeployV2Base.s.sol

```diff
@@ -28,59 +41,269 @@ abstract contract DeployV2Base is Script, ConfigBase {
+    /// @param salt Salt for deterministic deployment
+    /// @param creationCode Bytecode with constructor args
+    /// @return deployedAddr Address of the deployed or existing contract
+    function __deployContractIfNeeded(
+        string memory contractName,
+        uint64 chainId,
+        bytes32 salt,
+        bytes memory creationCode
+    )
+        internal
+        returns (address deployedAddr)
+    {
+        // Check if contract is already deployed
+        if (_isContractDeployed(chainId, contractName)) {
+            ContractStatus memory status = _getContractStatus(chainId, contractName);
+            console2.log("[!] %s already deployed, skipping...", contractName);
+            console2.log("      at:", status.contractAddress);
+            contractAddresses[chainId][contractName] = status.contractAddress;
+            _exportContract(contractName, status.contractAddress, chainId);
+            return status.contractAddress;
+        }
+
+        // Deploy the contract if not already deployed
+        return __deployContract(contractName, chainId, salt, creationCode);
+    }
+
+    /// @notice Check if a contract is deployed using bytecode from locked artifacts
+    /// @param contractName Name of the contract
+    /// @param salt Salt used for deployment
+    /// @param args Constructor arguments (empty if none)
+    /// @return isDeployed Whether the contract is deployed
+    /// @return contractAddr Address of the contract
+    function __checkContract(
+        string memory contractName,
+        bytes32 salt,
+        bytes memory args
+    )
+        internal
+        returns (bool isDeployed, address contractAddr)
+    {
+        return __checkContractOnChain(contractName, salt, args, uint64(block.chainid));
+    }
+
+    /// @notice Check if a contract is deployed on a specific chain
+    /// @param contractName Name of the contract
+    /// @param salt Salt used for deployment
+    /// @param args Constructor arguments (empty if none)
+    /// @param chainId Chain ID to check
+    /// @return isDeployed Whether the contract is deployed
+    /// @return contractAddr Address of the contract
+    function __checkContractOnChain(
+        string memory contractName,
+        bytes32 salt,
+        bytes memory args,
+        uint64 chainId
+    )
+        internal
+        returns (bool isDeployed, address contractAddr)
+    {
+        // Get bytecode from locked artifacts (Nexus style)
+        string memory artifactPath = string(abi.encodePacked("script/locked-bytecode/", contractName, ".json"));
+        bytes memory bytecode = vm.getCode(artifactPath);
+
+        // Compute address
+        contractAddr = DeterministicDeployerLib.computeAddress(bytecode, args, salt);
+
+        // Check if deployed using assembly (Nexus style)
+        uint256 codeSize;
+        assembly {
+            codeSize := extcodesize(contractAddr)
+        }
+
+        isDeployed = codeSize > 0;
```

Comment:
same comment as before, assembly is not necessary here. directly use length based comparision